### PR TITLE
JENA-2017: Annotation syntax for SPARQL*

### PR DESCRIPTION
**Pull request by afs**
Includes various "tidy up" commits accumulated while this branch was open.

### CODE REVIEW

**Comment by afs** [comment]
Detail on https://issues.apache.org/jira/browse/JENA-2017 .

File: jena-arq/src/main/java/org/apache/jena/riot/RDFParserBuilder.java [hunk 1]
```diff
@@ -612,10 +612,17 @@ public RDFParser build() {
...
+            if ( baseUri == null )
+                baseUri = IRILib.filenameToIRI(path.toString());
+        } else {
+            if ( baseUri == null && uri != null ) {
+                if ( resolver != null )
+                    baseUri = resolver.resolveToString(uri);
+                else
+                    baseUri = IRIResolver.resolveString(uri);
+            }
+        }
```
Comment (kinow) [inline jena-arq/src/main/java/org/apache/jena/riot/RDFParserBuilder.java:625]: Or maybe

```java
if ( baseUri == null ) {
    if ( path != null ) {
        baseUri = IRILib.filenameToIRI(path.toString());
    } else if ( uri != null ) {
        baseUri = resolver != null ? resolver.resolveToString(uri) : IRIResolver.resolveString(uri);
    }
}
```

I think it should eval the same way, but moving the common condition `baseUri == null` out?

File: jena-arq/src/main/java/org/apache/jena/riot/web/HttpNames.java [hunk 2]
```diff
@@ -63,21 +63,22 @@
...
     public static final String paramTarget          = "target" ;
-    
-    
+
+
     public static final String paramStyleSheet      = "stylesheet" ;
     public static final String paramAccept          = "accept" ;
-    public static final String paramOutput1         = "output" ;        // See Yahoo! developer: http://developer.yahoo.net/common/json.html 
-    public static final String paramOutput2         = "format" ;        // Alternative name 
+    public static final String paramOutput1         = "output" ;        // See Yahoo! developer: http://developer.yahoo.net/common/json.html
```
Comment (kinow) [inline jena-arq/src/main/java/org/apache/jena/riot/web/HttpNames.java:70]: For later maybe; this link is 404. Int he wayback machine several 302 entries. The 302'd page was 404. Last good crawled page for the original link I found is from 2006: http://web.archive.org/web/20060324091551/http://developer.yahoo.com/common/json.html

File: jena-arq/src/main/java/org/apache/jena/sparql/lang/ParserBase.java [hunk 3]
```diff
@@ -399,7 +400,16 @@ protected Node createBNode(String label, int line, int column) {
...
         return activeLabelMap.asNode(label) ;
     }
-
+    
+    protected Node preConditionAnnotation(Node s, Node p, Path path, Node o, int line, int column) {
+        if ( p != null )
+            return p;
+        if ( path instanceof P_Link )
+            return ((P_Link)path).getNode();
+        throwParseException("Only a simple paths allowed with annotation syntax", line, column) ;
```
Comment (kinow) [inline jena-arq/src/main/java/org/apache/jena/sparql/lang/ParserBase.java:409]: s/simple paths/simple path?

File: jena-arq/src/test/java/org/apache/jena/arq/junit/sparql/tests/QuerySyntaxTest.java [hunk 4]
```diff
@@ -30,17 +30,20 @@
 
 public class QuerySyntaxTest implements Runnable {
     final boolean       expectLegalSyntax;
+    // Rwquired syntax, null for by file extension.
```
Comment (kinow) [inline jena-arq/src/test/java/org/apache/jena/arq/junit/sparql/tests/QuerySyntaxTest.java:33]: s/Rwquired/Required

**Review by kinow — APPROVED** [review]
:+1: good idea to use RDF-star in the code, instead of RDF*. For some reason in my mind I read RDF* as RDF-asterisk first, then remember that it's RDF-star (probably due to Portuguese). Some minor comments, but no blockers. And happy new year!

File: jena-arq/src/main/java/org/apache/jena/riot/RDFParserBuilder.java (hunk 1 above)
Comment (afs) [reply jena-arq/src/main/java/org/apache/jena/riot/RDFParserBuilder.java:625]: Yes, much better!

File: jena-arq/src/main/java/org/apache/jena/riot/web/HttpNames.java (hunk 2 above)
Comment (afs) [reply jena-arq/src/main/java/org/apache/jena/riot/web/HttpNames.java:70]: Good find. Reading the link, not sure the comment adds much - "output" is for any syntax nowadays. Removed.

**Pull request merged**